src/renderer/task-runner.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  BisectRequest,
+  ElectronReleaseChannel,
+  RunResult,
+  RunnableVersion,
+  SetFiddleOptions,
+  SetupRequest,
+  TestRequest,
+} from '../interfaces';
+import { IpcEvents } from '../ipc-events';
+
+import { App } from './app';
+import { AppState } from './state';
+import { IpcRendererManager } from './ipc';
+import { Runner } from './runner';
+
+import { getVersionRange } from '../utils/get-version-range';
+import { normalizeVersion } from '../utils/normalize-version';
+
+export class TaskRunner {
+  private readonly autobisect: (v: RunnableVersion[]) => Promise<RunResult>;
+  private readonly done: (r: RunResult) => void;
+  private readonly hide: (channels: ElectronReleaseChannel[]) => Promise<void>;
+  private readonly log: (message: string) => void;
+  private readonly open: (o: SetFiddleOptions) => Promise<void>;
+  private readonly run: () => Promise<RunResult>;
+  private readonly setVersion: (ver: string) => Promise<void>;
+  private readonly show: (channels: ElectronReleaseChannel[]) => Promise<void>;
+
+  constructor(
+    app: App,
+    private readonly appState: AppState,
+    runner: Runner,
```

Comment:
Could we call `app.runner` here rather than using a new parameter?

Reply:
e89536e82ff48932d1b142fa35b897735c7c7190